analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -168,133 +144,275 @@ public static void LoopWithBreak(System.Collections.Generic.IEnumerable<string>
     }
 }
 
-public interface IWithDefaultImplementation
+class NullableOfCustomTypes
 {
-    int DoSomething()
+    void Assignment(AStruct? nullable)
     {
-        int? i = null;
-        return i.Value; //FIXME Non-compliant
+        _ = nullable.Value;                                              // Compliant, unknown
+        _ = (AStruct)nullable;                                           // Compliant, unknown
+        _ = new AStruct?(nullable.Value).Value;
+
+        nullable = null;
+        _ = nullable.Value;                                              // Noncompliant
+        nullable = new AStruct();
+        _ = nullable.Value;
+        nullable = new AStruct?(new AStruct());
+        _ = nullable.Value;
+
+        nullable = new AStruct?();
+        _ = nullable.Value;                                              // Noncompliant
+        nullable = new Nullable<AStruct>();
+        _ = nullable.Value;                                              // Noncompliant
+
+        _ = (new Nullable<AStruct>()).Value;                             // Noncompliant
+        _ = (null as Nullable<AStruct>).Value;                           // Noncompliant
+
+        nullable = null;
+        _ = ((AStruct?)nullable).Value;                                  // Noncompliant
+        _ = (nullable as AStruct?).Value;                                // FN
+        _ = ((AStruct?)(nullable as AStruct?)).Value;                    // FN
```

Comment:
Please re-assign `nullable=null` before each line to make sure we didn't learn the NotNull from using `.Value`

Reply:
More methods just adds more scaffolding in this case.

Reply:
Removed the additional methods